src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -48,18 +49,20 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
     {
         $resources = $this->resourcesResolver->getResources($requestConfiguration, $repository);
 
-        if ($resources instanceof Pagerfanta) {
+        $data = $resources instanceof ResourceGridView ? $resources->getData() : $resources;
```

Comment:
`$paginator` is misleading as it may be just an `iterable` instead of paginator.

I think we can leave it now as it is, but the real issue is `$this->resourcesResolver->getResources()` returning `mixed`.